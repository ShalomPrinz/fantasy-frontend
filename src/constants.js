export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 800;
