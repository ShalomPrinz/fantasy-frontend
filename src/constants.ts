export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 1000;
export const FULL_SCREEN_MIN_WIDTH = 1200;

// My Backend Endpoints
export const baseURL = "http://localhost:8080";
export const playersUrl = "/players/query";
export const usernamesUrl = "/users/query";
export const registerUrl = "/register";
export const userInfoUrl = "/user";
export const addPlayerUrl = "/user/addplayer";
export const removePlayerUrl = "/user/removeplayer";
export const acceptLeagueInviteUrl = "/accept/leagueinvite";
export const rejectLeagueInviteUrl = "/reject/leagueinvite";
