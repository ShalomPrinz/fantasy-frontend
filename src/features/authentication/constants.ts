/* API Endpoints */

export const registerUrl = "/register";
export const userInfoUrl = "/user";
