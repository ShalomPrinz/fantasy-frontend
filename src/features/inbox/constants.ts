/* API Endpoints */

export const acceptLeagueInviteUrl = "/accept/leagueinvite";
export const rejectLeagueInviteUrl = "/reject/leagueinvite";
