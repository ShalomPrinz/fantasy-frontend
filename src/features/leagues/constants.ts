export const LEAGUE_ID_LENGTH = 20;
export const LEAGUE_TEAM_LAYOUT_MAX_WIDTH = 680;

/* API Endpoints */

export const createLeagueUrl = "/newleague";
export const inviteLeagueMemberUrl = "/leagueinvite";
export const leagueInfoUrl = "/league";
