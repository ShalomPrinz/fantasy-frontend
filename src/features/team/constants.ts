export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 1000;

/* API Endpoints */

export const addPlayerUrl = "/user/addplayer";
export const removePlayerUrl = "/user/removeplayer";
