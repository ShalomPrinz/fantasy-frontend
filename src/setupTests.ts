// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import axios from "axios";
import { setupServer } from "msw/node";

import { handlers } from "../test";
import * as TeamContext from "./contexts/TeamContext";
import * as UserContext from "./features/authentication/contexts/UserContext";
import { LeagueInfo, Message, Player, Team, User } from "./types";

jest.setTimeout(10000);

axios.defaults.baseURL = "";

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

interface MockedUserContext {
  logout?: () => Promise<void>;
  state?: UserContext.UserState;
  user?: User;
}

interface MockedUser {
  inbox?: Message[];
  leagues?: LeagueInfo[];
  name?: string;
  players?: Player[];
}

export const getTestUser = (user?: MockedUser): User => ({
  id: "id",
  inbox: user?.inbox || [],
  leagues: user?.leagues || [],
  name: user?.name || "",
  team: new Team(user?.players || []),
});

export const mockUser = (context: MockedUserContext) => {
  const userContext = jest.spyOn(UserContext, "useUser");

  const { logout, state, user } = context;
  const mockedLogout = typeof logout !== "undefined" ? logout : async () => {};
  const mockedState =
    typeof state !== "undefined" ? state : UserContext.UserState.LOGGED_USER;
  const mockedUser = typeof user !== "undefined" ? user : getTestUser();

  userContext.mockImplementation(() => ({
    loading: false,
    login: async () => {},
    logout: mockedLogout,
    register: async () => {},
    state: mockedState,
    user: mockedUser,
    invalidate: async () => {},
  }));

  return userContext.mockRestore;
};

export const mockTeam = (team: Team) => {
  const teamContext = jest.spyOn(TeamContext, "useTeamState");
  teamContext.mockImplementation(() => team);
};

export * from "../test";
