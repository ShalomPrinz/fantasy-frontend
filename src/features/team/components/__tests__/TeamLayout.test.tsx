import { mockTeam, render, setProperty, setWindowSize } from "setupTests";

import { TeamLayout } from "../";
import { Team } from "../../";
import { FIELD_LAYOUT_MIN_WIDTH } from "../../constants";
import * as constants from "../../constants";
import { missingIdPlayer, missingTeamPlayer, player } from "./players";

describe("TeamLayout", () => {
  it("should render TeamList component", () => {
    const team = new Team([player]);
    mockTeam(team);

    setWindowSize(FIELD_LAYOUT_MIN_WIDTH - 1);
    const { asFragment } = render(<TeamLayout />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render TeamLayout component", () => {
    const team = new Team([player]);
    mockTeam(team);

    setWindowSize(FIELD_LAYOUT_MIN_WIDTH);
    const { asFragment } = render(<TeamLayout />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render bigger row margin if field width is higher than 600px", () => {
    const team = new Team([player]);
    mockTeam(team);

    setWindowSize(FIELD_LAYOUT_MIN_WIDTH);
    setProperty(constants, "FIELD_IMAGE_DEFAULT_WIDTH", 601);
    const { asFragment } = render(<TeamLayout />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe("when window resize", () => {
    it("should resize player image", () => {
      // TODO. I couldn't find any test library that support that
    });
  });

  describe("Runtime Errors", () => {
    it("should render with damaged data: no player team", () => {
      //@ts-expect-error should miss team property
      const team = new Team([missingTeamPlayer]);
      mockTeam(team);

      const { asFragment } = render(<TeamLayout />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should render with damaged data: no player id", () => {
      //@ts-expect-error should miss id property
      const team = new Team([missingIdPlayer]);
      mockTeam(team);

      const { asFragment } = render(<TeamLayout />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should cause uncaught error and render Error Boundary", () => {
      //@ts-expect-error useTeamState is undefined and TeamLayout calls it as a function
      mockTeam(undefined);

      const { asFragment } = render(<TeamLayout />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
