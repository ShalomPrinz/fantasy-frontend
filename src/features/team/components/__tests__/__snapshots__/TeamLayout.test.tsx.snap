// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamLayout Runtime Errors should cause uncaught error and render Error Boundary 1`] = `
<DocumentFragment>
  <div
    class="text-center my-3 p-4 bg-danger text-white w-75 mx-auto"
  >
    <h2>
      Team Layout Not Available
    </h2>
    <button
      class="fs-4 bg-white rounded"
      type="button"
    >
      Refresh
    </button>
  </div>
</DocumentFragment>
`;

exports[`TeamLayout Runtime Errors should render with damaged data: no player id 1`] = `
<DocumentFragment>
  <div
    class="position-relative"
  >
    <img
      alt="Team Field Background"
      class="w-100 centered-image rounded"
      src="field.png"
      title="Team Field Background"
    />
    <div
      class="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
    >
      <div
        class="my-3 row"
      >
        <div
          class="mx-auto col"
        >
          <div
            class="text-center"
            style="width: 84.14px;"
          >
            <img
              class=""
              src="Barcelona.png"
              style="width: 60.1px;"
              title="Barcelona"
            />
            <div
              class="text-truncate bg-default rounded p-1 w-100"
            >
              Ter Stegen
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TeamLayout Runtime Errors should render with damaged data: no player team 1`] = `
<DocumentFragment>
  <div
    class="position-relative"
  >
    <img
      alt="Team Field Background"
      class="w-100 centered-image rounded"
      src="field.png"
      title="Team Field Background"
    />
    <div
      class="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
    >
      <div
        class="my-3 row"
      >
        <div
          class="mx-auto col"
        >
          <div
            class="text-center"
            style="width: 84.14px;"
          >
            <img
              class=""
              src="Error.png"
              style="width: 60.1px;"
            />
            <div
              class="text-truncate bg-default rounded p-1 w-100"
            >
              Ter Stegen
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TeamLayout should render TeamLayout component 1`] = `
<DocumentFragment>
  <div
    class="position-relative"
  >
    <img
      alt="Team Field Background"
      class="w-100 centered-image rounded"
      src="field.png"
      title="Team Field Background"
    />
    <div
      class="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
    >
      <div
        class="my-1 row"
      >
        <div
          class="mx-auto col"
        >
          <div
            class="text-center"
            style="width: 77px;"
          >
            <img
              class=""
              src="Barcelona.png"
              style="width: 55px;"
              title="Barcelona"
            />
            <div
              class="text-truncate bg-default rounded p-1 w-100"
            >
              Ter Stegen
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-1 row"
      />
      <div
        class="my-1 row"
      />
      <div
        class="my-1 row"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TeamLayout should render TeamList component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="w-50 col"
      >
        <h1
          class="centered-flex pt-4"
        >
           GK 
        </h1>
        <div
          class="d-flex align-items-center ps-5"
        >
          <img
            class=""
            src="Barcelona.png"
            style="width: 60px;"
            title="Barcelona"
          />
          <h4
            class="text-truncate ps-2 my-0"
          >
             Ter Stegen 
          </h4>
        </div>
      </div>
      <div
        class="w-50 col"
      >
        <h1
          class="centered-flex pt-4"
        >
           DEF 
        </h1>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="w-50 col"
      >
        <h1
          class="centered-flex pt-4"
        >
           MID 
        </h1>
      </div>
      <div
        class="w-50 col"
      >
        <h1
          class="centered-flex pt-4"
        >
           ATT 
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TeamLayout should render bigger row margin if field width is higher than 600px 1`] = `
<DocumentFragment>
  <div
    class="position-relative"
  >
    <img
      alt="Team Field Background"
      class="w-100 centered-image rounded"
      src="field.png"
      title="Team Field Background"
    />
    <div
      class="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
    >
      <div
        class="my-3 row"
      >
        <div
          class="mx-auto col"
        >
          <div
            class="text-center"
            style="width: 84.14px;"
          >
            <img
              class=""
              src="Barcelona.png"
              style="width: 60.1px;"
              title="Barcelona"
            />
            <div
              class="text-truncate bg-default rounded p-1 w-100"
            >
              Ter Stegen
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
      <div
        class="my-3 row"
      />
    </div>
  </div>
</DocumentFragment>
`;
